src/main/java/athleticli/ui/Parser.java

```diff
@@ -253,4 +271,76 @@ public static Swim.SwimmingStyle parseSwimmingStyle(String swimmingStyle) throws
         }
     }
 
+
+    public static AddSleepCommand parseSleepAdd(String commandArgs) throws AthletiException {
+
+        final String STARTMARKER = "/start";
+        final String ENDMARKER = "/end";
+
+        int startMarkerPos = commandArgs.indexOf(STARTMARKER);
+        int endMarkerPos = commandArgs.indexOf(ENDMARKER);
+
+        if (startMarkerPos == -1 || endMarkerPos == -1) {
+            throw new AthletiException("Please specify both the start and end time of your sleep.");
+        }
+
+        if (startMarkerPos > endMarkerPos) {
+            throw new AthletiException("Please specify the start time of your sleep before the end time.");
+        }
+    
+        String startTime = commandArgs.substring(startMarkerPos + STARTMARKER.length(), endMarkerPos).trim();
+        String endTime = commandArgs.substring(endMarkerPos + ENDMARKER.length()).trim();
+
+        if(startTime.isEmpty() || endTime.isEmpty()) {
+            throw new AthletiException("Please specify both the start and end time of your sleep.");
+        }
+
+        return new AddSleepCommand(startTime, endTime);
+    }
+
+    public static DeleteSleepCommand parseSleepDelete(String commandArgs) throws AthletiException {
+        int index;
+
+        try {
+            index = Integer.parseInt(commandArgs.trim());
+        } catch (NumberFormatException e) {
+            throw new AthletiException("Please specify the index of the sleep record you want to delete.");
+        }
+
+        return new DeleteSleepCommand(index);
+    }
+
+    public static EditSleepCommand parseSleepEdit(String commandArgs) throws AthletiException {
+        final String STARTMARKER = "/start";
+        final String ENDMARKER = "/end";
+
+        int startMarkerPos = commandArgs.indexOf(STARTMARKER);
+        int endMarkerPos = commandArgs.indexOf(ENDMARKER);
+
+        int index;
+
+        if (startMarkerPos == -1 || endMarkerPos == -1) {
+            throw new AthletiException("Please specify both the start and end time of your sleep.");
+        }
+
+        if (startMarkerPos > endMarkerPos) {
+            throw new AthletiException("Please specify the start time of your sleep before the end time.");
+        }
+
+        try {
+            index = Integer.parseInt(commandArgs.substring(0, startMarkerPos).trim());
+        } catch (NumberFormatException e) {
+            throw new AthletiException("Please specify the index of the sleep record you want to edit.");
+        }
```

Comment:
Good exception handling! You could use the Message class to separate the user notifications from the parser logic.

Reply:
Thanks! Will be implemented in the next cycle.